Automattic/jetpack
# Access modal mixed approach

Works with D126720-code

Task list:
- [x] Do not use WPCOM page to generate the magic link 
- [x] Fix issue with multiple subscribe buttons
- [x] Replace "code" screen  with an "email" input (if not provided) 
- [x] Send a magic link in the background from WPCOM side (sent but needs redirect to WPCOM
- [ ] (WPCOM repo) Add an endpoint to redirect with token to the post/page from the redirect
- [ ] (Jetpack repo) Add an endpoint (or reuse the existing one) that set the new token in session/cookie on instance backend and closes the tab.
- [ ] (Jetpack ) Add an endpoint that accepts only calls from the front-end and accepts a "token" parameter and checks it against the backend one `$_COOKIE['token']`
- [ ] (Jetpack js) Create a JS polling routine that checks for token changes, and reload the page if token is changed.

## Proposed changes:

* See: p81Rsd-1vQ-p2#comment-1620
* Keep user in-site when getting subscriptions access
* streamline login process without opening new tab
* Can potentially be segmented for WPCOM users (using regular WPCOM email-less login)  vs self-hosted users 

## Testing instructions:






* Apply D126720-code to your sandbox
* For Simple sites:
  * Apply this PR on top (if working with Simple site)
* For Jetpack
  * Create a JN instance and link to your sandbox
  * Link Jetpack
* Setup Monetize and create tier plans
* Create a post with a paywall plan
* Click "Login"/"Get Access"
You should see something like this:

Uploading Screen Recording 2023-10-26 at 12.28.41.mov…


* Clicking on the email link open and close a new tab quickly, and reload the browsed page
